redfish-core/lib/log_services.hpp

```diff
@@ -6326,5 +6368,923 @@ inline void requestRoutesFullAuditLogDownload(App& app)
+        const std::string uriIdPattern{"<str>"};
+        size_t uriIdPos = redfishUri.rfind(uriIdPattern);
+        if (uriIdPos == std::string::npos)
+        {
+            if (entryJsonIdx > 0)
+            {
+                asyncResp->res.jsonValue["Members"][entryJsonIdx - 1]["Links"]
+                                        ["OriginOfCondition"] = {
+                    {"@odata.id", redfishUri}};
+            }
+            else
+            {
+                asyncResp->res.jsonValue["Links"]["OriginOfCondition"] = {
+                    {"@odata.id", redfishUri}};
+            }
+            return;
+        }
+
+        // Fill the all parents Redfish URI id.
+        // For example, the processors id for the core.
+        // "/redfish/v1/Systems/system/Processors/<str>/SubProcessors/core0"
+        crow::connections::systemBus->async_method_call(
+            [asyncResp, dbusObjPath, entryJsonIdx, redfishUri, uriIdPos,
+             uriIdPattern](
+                const boost::system::error_code ec1,
+                const boost::container::flat_map<
+                    std::string, boost::container::flat_map<
+                                     std::string, std::vector<std::string>>>&
+                    subtree) mutable {
+            if (ec1)
+            {
+                BMCWEB_LOG_ERROR(
+                    "DBUS response error [{} : {}] when tried to fill the parent objects id in the RedfishURI: {} of the isolated hareware: {}",
+                    ec1.value(), ec1.message(), redfishUri, dbusObjPath.str);
+
+                messages::internalError(asyncResp->res);
+                return;
+            }
+
+            while (uriIdPos != std::string::npos)
+            {
+                std::string parentRedfishUri = redfishUri.substr(0,
+                                                                 uriIdPos - 1);
+                RedfishUriListType::const_iterator parentRedfishUriIt =
+                    std::find_if(redfishUriList.begin(), redfishUriList.end(),
+                                 [parentRedfishUri](const auto& ele) {
+                    return parentRedfishUri == ele.second;
+                });
+
+                if (parentRedfishUriIt == redfishUriList.end())
+                {
+                    BMCWEB_LOG_ERROR(
+                        "Failed to fill Links:OriginOfCondition because unable to get parent Redfish URI [{}] DBus interface for the identified Redfish URI: {} of the given DBus object path: {} ",
+                        parentRedfishUri, redfishUri, dbusObjPath.str);
+                    messages::internalError(asyncResp->res);
+                    return;
+                }
+
+                std::string parentObj;
+                for (const auto& obj : subtree)
+                {
+                    if (dbusObjPath.str.find(
+                            sdbusplus::message::object_path(obj.first)
+                                .filename()) == std::string::npos)
```

Comment:
Instead of searching a pattern from `dbusObjPath`, is this trying to match a file name?

```
                    if (dbusObjPath.filename() != 
                                sdbusplus::message::object_path(obj.first) .filename())
```